addon-test-support/-private/utils.js

```diff
@@ -0,0 +1,14 @@
+import Ember from 'ember';
+
+export function checkMatcher(message, matcher) {
+  if (typeof matcher === 'string') {
+    return message.includes(matcher);
+  } else if (matcher instanceof RegExp) {
+    return !!message.match(matcher);
+  } else if (matcher) {
+    Ember.assert(`ember-qunit-assert-helpers can only match Strings and RegExps. "${typeof matcher}" was provided.`);
```

Comment:
Probably best to `throw new Error` here. Using `Ember.assert` while we are mocking it feels weird to me...